src/compile/layer.ts

```diff
@@ -0,0 +1,228 @@
+import {ScaleComponent} from './scale';
+import {ScaleType} from '../scale';
+import {VgData, VgAxis, VgLegend} from '../vega.schema';
+
+
+export class LayerModel extends Model {
+  private _children: UnitModel[];
+
+  constructor(spec: LayerSpec, parent: Model, parentGivenName: string) {
+    super(spec, parent, parentGivenName);
+
+    this._config = this._initConfig(spec.config, parent);
+    this._children = spec.layers.map((layer, i) => {
+      // we know that the model has to be a unit model beacuse we pass in a unit spec
+      return buildModel(layer, this, this.name('layer_' + i)) as UnitModel;
+    });
+  }
+
+  private _initConfig(specConfig: Config, parent: Model) {
+    return mergeDeep(duplicate(defaultConfig), specConfig, parent ? parent.config() : {});
+  }
+
+  public has(channel: Channel): boolean {
+    // layer does not have any channels
+    return false;
+  }
+
+  public children() {
+    return this._children;
+  }
+
+  private hasSummary() {
+    // TODO: don't just use the first child
+    const summary = this._children[0].component.data.summary;
+    for (let i = 0; i < summary.length; i++) {
+      if (keys(summary[i].measures).length > 0) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public isOrdinalScale(channel: Channel) {
+    // since we assume shared scales we can just ask the first child
+    return this._children[0].isOrdinalScale(channel);
+  }
+
+  public dataTable(): string {
+    // TODO: think about what this should return, used by layout
+    return (this.hasSummary() ? SUMMARY : SOURCE) + '';
+  }
+
+  public fieldDef(channel: Channel): FieldDef {
+    return null; // layer does not have field defs
+  }
+
+  public stack() {
+    return null; // this is only a property for UnitModel
+  }
+
+  public parseData() {
+    this._children.forEach((child) => {
+      child.parseData();
+    });
+    this.component.data = parseLayerData(this);
+    console.log(this.component.data);
+  }
+
+  public parseSelectionData() {
+    // TODO: @arvind can write this
+    // We might need to split this into compileSelectionData and compileSelectionSignals?
+  }
+
+  public parseLayoutData() {
+    // TODO: correctly union ordinal scales rather than just using the layout of the first child
+    this._children.forEach((child, i) => {
+      child.parseLayoutData();
+    });
+    this.component.layout = parseLayerLayout(this);
+  }
+
+  public parseScale() {
+    const model = this;
+
+    let scaleComponent = this.component.scale = {} as Dict<ScaleComponent[]>;
+
+    this._children.forEach(function(child) {
+      child.parseScale();
+
+      // TODO: correctly implement independent scale
+      if (true) { // if shared/union scale
+        keys(child.component.scale).forEach(function(channel) {
+          let newScales = child.component.scale[channel] as ScaleComponent[];
+          if (!newScales) {
+            return;
+          }
+
+          const existingScales = scaleComponent[channel];
```

Comment:
`existingScales` => `scales` or `modelScales` or something that suggests that it is this model's.